### PR TITLE
test(sirx): create sir-x tests

### PR DESCRIPTION
**Pull request by felipehuerta17**
In response to #64, I begin writing SIR-X tests from the sir template that @jia200x created.

As SIR-X is mathematically more complex, the logical tests will be adapted but new mathematical tests will be implemented.

I will not test any .fit() functionality here as that will be a separate PR dealing with a potentially much more difficult issue.

### CODE REVIEW

File: tests/test_sirx.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        infected_end = model.fetch()[-1, 4]
+
+        assert abs(infected_end - 2524.388983) < 0.0001
+
+    def test_expconv(self, model):
+        # Check that the numerical solution of the
+        # reduction from sir-x to sir
+        # converges to I ~ I_0 * exp (alpha-beta)
+        # as (S,t) -> (1,0)
+        # Asume small amount of infected
```
Comment (sasalatart) [inline tests/test_sirx.py:44]:
```suggestion
        # Assume small amount of infected
```

File: tests/test_sirx.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        n_i_0 = 1
+        # Asume large population
```
Comment (sasalatart) [inline tests/test_sirx.py:46]:
```suggestion
        # Assume large population
```

File: tests/test_sirx.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        initial_conds = [GUANGDONG_IC[0], n_i_0, 0, n_i_0]
+        # Asume very short time
```
Comment (sasalatart) [inline tests/test_sirx.py:48]:
```suggestion
        # Assume very short time
```

File: tests/test_sirx.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        t = 1 / (24 * 3600)
+        # Make kappa, kappa_0 and
```
Comment (sasalatart) [inline tests/test_sirx.py:50]: That final "and" is a bit strange, or is it a "connector" with "Calculate analytical..."?

File: tests/test_sirx.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        sirx_red_params = [DEFAULT_PARAMS[0], DEFAULT_PARAMS[1], 0, 0, 0]
+        # Calculate analytical and numerical solutions
+        model.set_params(sirx_red_params, initial_conds)
+        model.solve(t, 2)
+        n_inf_num = model.fetch()[-1, 1]
+        n_inf_analytical = n_inf_num * np.exp(
+            (DEFAULT_PARAMS[0] - DEFAULT_PARAMS[1]) * t
+        )
+        perc_error = abs(n_inf_analytical - n_inf_num) / n_inf_num
+        # Asert over the difference < 1e-4
```
Comment (sasalatart) [inline tests/test_sirx.py:60]:
```suggestion
        # Assert over the difference < 1e-4
```

File: tests/test_sirx.py (hunk 4 above)
Comment (felipehuerta17) [reply tests/test_sirx.py:50]: Well spotted! the full string was:

        # Make kappa, kappa_0 and inf_over_test zero to reduce
        # the SIR-X to SIR
But for some reason my brain got disconnected halfway through

**Comment by felipehuerta17** [comment]
@sasalatart comments implemented and now we have tests for SIR-X!
image

**Pull request merged**